File: Engine/Tools/WismutCLI/CommandLineParser.cs
namespace WismutCLI
{
	public class CommandParameters
	{
		public string Command = "";
		public Dictionary<string, string> Arguments = new();
	}

	public static class CommandLineParser
	{
		public static CommandParameters Parse(string[] args)
		{
			var options = new CommandParameters();
			foreach (var arg in args)
			{
				if (!arg.StartsWith("--"))
					options.Command = arg;
				else
				{
					string[] strings = arg.Split("=");
					string key = strings[0].Substring(2, strings[0].Length - 2);
					string value = "";
					if (strings.Length > 1)
						value = strings[1];
					options.Arguments.Add(key, value);
				}
			}

			return options;
		}
	}
}

File: Engine/Source/CLI/Source/CommandLineParser.cpp
#include "CommandLineParser.h"
#include <sstream>

CommandParameters CommandLineParser::Parse(int argc, char* argv[])
{
	CommandParameters result;

	for (int i = 1; i < argc; ++i)
	{
		std::string arg = argv[i];

		if (arg.starts_with("--"))
		{
			size_t eqPos = arg.find('=');
			if (eqPos != std::string::npos)
			{
				std::string key = arg.substr(2, eqPos - 2);
				std::string value = arg.substr(eqPos + 1);
				result.Arguments[key] = value;
			}
			else
			{
				std::string key = arg.substr(2);
				result.Arguments[key] = "";
			}
		}
		else
		{
			if (result.Сommand.empty())
				result.Сommand = arg;
		}
	}

	return result;
}
